File: Iteration 1/Source Code/Person.java
import java.util.Date;

public abstract class Person {
    protected  String name;
    protected  String surname;
    protected  Date birthdate;
    protected  char gender;

    public Person(){

    }
    public Person(String name, String surname){
        this.name = name;
        this.surname = surname;
    }
    public Person(String name, String surname, Date birthdate, char gender){
        this.name = name;
        this.surname = surname;
        this.birthdate = birthdate;
        this.gender = gender;
    }

    public String getName(){
        return name;

    };

    public String getSurname(){
        return surname;
    };

    public Date getBirthdate() {
        return birthdate;
    }

    public char getGender() {
        return gender;
    }
}

File: Iteration_3/Person.py
from datetime import date
from abc import ABC, abstractmethod

class Person(ABC):
    def __init__(self, _name: str = None, _surname: str = None, _birthdate: date = None, _gender: str = None, _ssn: str = None):
        self._name = _name
        self._surname = _surname
        self._birthdate = _birthdate
        self._gender = _gender
        self._ssn = _ssn

    @abstractmethod
    def get_name(self):
        """Get the name of the person"""
        pass

    @abstractmethod
    def get_surname(self):
        """Get the surname of the person"""
        pass

    @abstractmethod
    def get_birthdate(self):
        """Get the birthdate of the person"""
        pass

    @abstractmethod
    def get_gender(self):
        """Get the gender of the person"""
        pass
    @abstractmethod
    def initialize_interface(self, interface):
        pass

    def get_id(self):
        return self._ssn
        
